src/app/workflow/launch-third-party/launch-third-party.component.ts

```diff
@@ -279,37 +277,12 @@ export class LaunchThirdPartyComponent extends Base implements OnChanges, OnInit
     //     this.usersCloudInstances = cloudInstances;
     //   });
     // }
-
-    this.gA4GHFilesQuery
-      .getToolFiles(this.descriptorTypeCompatService.stringToDescriptorType(this.workflow.descriptorType), [
-        FileTypeEnum.PRIMARYDESCRIPTOR,
-        FileTypeEnum.SECONDARYDESCRIPTOR,
-      ])
-      .pipe(takeUntil(this.ngUnsubscribe))
-      .subscribe((fileDescriptors) => {
-        if (fileDescriptors && fileDescriptors.length) {
-          // No idea if this.workflow.descriptorType is the one that's required or if it's some other enum
-          const descriptorType = this.workflow.descriptorType;
-          const descriptorLanguageEnum =
-            this.descriptorLanguageService.workflowDescriptorTypeEnumToExtendedDescriptorLanguageBean(
-              descriptorType
-            ).descriptorLanguageEnum;
-          this.workflowsService.primaryDescriptor1(this.workflow.id, descriptorType, this.selectedVersion.name).subscribe((sourceFile) => {
-            this.descriptorsService.updatePrimaryDescriptor(sourceFile);
-            if (fileDescriptors.some((file) => file.file_type === FileTypeEnum.SECONDARYDESCRIPTOR)) {
-              this.workflowsService
-                .secondaryDescriptors1(this.workflow.id, descriptorLanguageEnum, this.selectedVersion.name)
-                .subscribe((sourceFiles: Array<SourceFile>) => {
-                  this.descriptorsService.updateSecondaryDescriptors(sourceFiles);
-                });
-            }
-          });
-        }
-      });
   }
 
   ngOnChanges(changes: SimpleChanges): void {
     this.descriptorsQuery.clear();
+    this.primaryDescriptorSubscription?.unsubscribe();
```

Comment:
> These `unsubscribe`s are here in case the user switches versions navigating away from the page, correct?

The `unsubscribe`s are so we don't handle any in-process requests that are left over from the previous set of inputs.  Imagine one of those request gets delayed, and then we receive the response, and handle it.  It could "overwrite" the actions of a superceding request, which would be bad.

> Are you sure there won't be a memory leak when the component is destroyed? I don't think so, but double-checking. On the one hand, I think subscriptions are not garbage collected if the observable hasn't been completed, on the other hand, http requests complete as soon as they are done, and once an observable is done, the subscriber will be garbage collectable.

Supposedly, http-based observables complete themselves after the result arrives:
https://stackoverflow.com/questions/35042929/is-it-necessary-to-unsubscribe-from-observables-created-by-http-methods-to-avoid

> In any case, I think you should also add the `takeUntil(this.ngUnsubscribe)`. If the user navigates away from the page while these requests are still happening, they'll get cancelled.

However, this is a good idea.  Say a request gets delayed, and we navigate away from the component.  We don't want to change state later when the response finally arrives.  Added.